src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/custom.py

```diff
@@ -0,0 +1,260 @@
+        states = client.list_query_results_for_management_group(
+            policy_states_resource,
+            management_group_name,
+            query_options)
+    else:
+        states = client.list_query_results_for_subscription(
+            policy_states_resource,
+            subscription_id,
+            query_options)
+
+    return states.value
+
+
+def summarize_policy_states(
+        cmd,
+        client,
+        management_group_name=None,
+        resource_group_name=None,
+        resource=None,
+        resource_provider_namespace=None,
+        resource_type=None,
+        policy_set_definition_name=None,
+        policy_definition_name=None,
+        policy_assignment_name=None,
+        from_value=None,
+        to_value=None,
+        top_value=None,
+        filter_clause=None):
+
+    from azure.mgmt.policyinsights.models import QueryOptions
+
+    query_options = QueryOptions(
+        top=top_value,
+        from_property=from_value,
+        to=to_value,
+        filter=filter_clause)
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if policy_assignment_name is not None:
+        if resource_group_name is not None:
+            summary = client.summarize_for_resource_group_level_policy_assignment(
+                subscription_id,
+                resource_group_name,
+                policy_assignment_name,
+                query_options)
+        else:
+            summary = client.summarize_for_subscription_level_policy_assignment(
+                subscription_id,
+                policy_assignment_name,
+                query_options)
+    elif policy_definition_name is not None:
+        summary = client.summarize_for_policy_definition(
+            subscription_id,
+            policy_definition_name,
+            query_options)
+    elif policy_set_definition_name is not None:
+        summary = client.summarize_for_policy_set_definition(
+            subscription_id,
+            policy_set_definition_name,
+            query_options)
+    elif resource is not None:
+        if not is_valid_resource_id(resource):
+            resource = resource_id(
+                subscription=subscription_id,
+                resource_group=resource_group_name,
+                namespace=resource_provider_namespace,
+                type=resource_type,
+                name=resource)
+        summary = client.summarize_for_resource(
+            resource,
+            query_options)
+    elif resource_group_name is not None:
+        summary = client.summarize_for_resource_group(
+            subscription_id,
+            resource_group_name,
+            query_options)
+    elif management_group_name is not None:
+        summary = client.summarize_for_management_group(
+            management_group_name,
+            query_options)
+    else:
+        summary = client.summarize_for_subscription(
+            subscription_id,
+            query_options)
+
+    return summary.value[0]
```

Comment:
Why do we only return the first value in the list?

Reply:
The REST API path returns a list but the list always contains a single object (summary object).